### PR TITLE
Added in basic auth and no-negotiate options

### PR DESCRIPTION
**Pull request by btoneill**
Two big options for GSSAPI were not in the module, added them in.

### CODE REVIEW

**Comment by LukasAud** [comment]
Hi @raybellis, sorry for the delay in feedback. We will be moving this back to active PRs and, hopefully, someone on our team can look at this soon. 

Meanwhile, as per @david22swan request, could you give us a bit more context on this change? What behaviour are these options responsible for? Are these common options or something more on the specialised area of apache? Any information you can provide will help us make a decision faster. Thank you for your patience.

**Comment by raybellis** [comment]
These options (except for the two `Auth*` ones are likely to be wanted by anyone using Kerberos via Basic Authentication (as opposed to requiring that the client browser pass a Kerberos ticket via GSSAPI).

File: templates/vhost/_gssapi.epp [hunk 1]
```diff
@@ -11,6 +14,12 @@
 <% if $localname { -%>
     GssapiLocalName <%= $localname %>
 <% } -%>
+<% if $negotiateonce { -%>
+    GssapiNegotiateOnce <%= $basicauth %>
```
Comment (david22swan) [inline templates/vhost/_gssapi.epp:18]: Small error, but should this not be:
```
GssapiNegotiateOnce <%= $negotiateonce %>
```

**Review by david22swan — CHANGES_REQUESTED** [review]
@btoneill Hey, sorry for the wait on review.

Anyway while the changes look good to me, aside from one small error I have noted, we would need some additional tests added in order to cover the new variables, the relevant test file being linked below:

- https://github.com/puppetlabs/puppetlabs-apache/blob/main/spec/defines/vhost_spec.rb

In addition while reviewing other apache PRs I found 3 others adding similar functionality to this one, one of which adds basicauth and has been merged in, i.e.:

- https://github.com/puppetlabs/puppetlabs-apache/pull/2212

Another that adds negotiate_once and has not yet been merged:

- https://github.com/puppetlabs/puppetlabs-apache/pull/2214

And a third that adds another 3 variables:

- https://github.com/puppetlabs/puppetlabs-apache/pull/2215

This is just a head's up so that you know to rebase your work and are aware that it may be fulfilled by another PR, depending on if it get's in a merge ready state before your own.

**Comment by david22swan** [comment]
@btoneill Hey, sorry but gonna close this PR as another has been merged in that adds the same functionality:
https://github.com/puppetlabs/puppetlabs-apache/pull/2214
Thanks for putting in the work though and I hope to see more from you in the future.

**Pull request closed without merging**